### ISSUE TITLE
Can't takeSnapshotAsync on Expo.GLView (Android)

### ISSUE THREAD

**Issue by pa-an** [expo/expo#1371]
I used `takeSnapshotAsync` on the View contain `Expo.GLView` component

- IOS: working.
- Android: Anything on screen is captured except GLView component.

Here is my example https://snack.expo.io/B1Mu3hZIz.
Could you please help me on this? if it is impossible, please suggest me another way to take image of `Expo.GLView`

**Comment by nikki93** [comment]
@pa-an unfortunately GLViews on Android aren’t snapshotted by takeSnapshotAsync — but we are adding a new snapshot function for GLViews in SDK 26. Will update when that’s released! We should also document this limitation of GLViews...

**Comment by pa-an** [comment]
Hi @nikki93, May I know when will SDK 26 release?

**Comment by yingdongzhang** [comment]
+1 Would love to know when it will be released.

**Comment by yingdongzhang** [comment]
@pa-an 

If you are in a rush and looking for an alternative, checkout this plugin:

https://github.com/kevinstumpf/react-native-signature-pad

It doesn't work with the latest react-native out of the box, but in the issues you'll find how to fix it.
I've successfully got it working on both iOS and Android. Hope it helps.

**Comment by tsapeta** [comment]
@pa-an @yingdongzhang 
We'd love to release it asap, however it doesn't depend on us. Thanks for your patience.

**Comment by pa-an** [comment]
@yingdongzhang Thank you very much, your solution worked for my problem 👍

**Comment by tsapeta** [comment]
@pa-an @yingdongzhang 
SDK 26 has been released, so you can now update Expo in your app and then do `glViewRef.takeSnapshotAsync()` which is designed for GLView and works on both platforms. See [docs](https://docs.expo.io/versions/latest/sdk/gl-view#takesnapshotasync) for more details. Feel free to ask if you have any questions :)

**Comment by yingdongzhang** [comment]
@tsapeta
That is awesome, thank you.

**Comment by taranda** [comment]
I am trying to figure out how I can take a snapShot of an `<ExpoPixi.Sketch>`.  It looks like this component renders an `<Expo.GLView>`, but I do not know how to invoke the `takeSnapshotAsync` method on the underlying `Expo.GLView`.  I tried to follow [this Expo blog post](https://blog.expo.io/drawing-signatures-with-expo-25d1629ca1ac) with no luck.

**Comment by tsapeta** [comment]
@taranda 
Oh, yeah, we need to update `expo-pixi` to make it possible to get a ref to underlying `GLView`. For simplicity I'll also add shortcut method directly to `ExpoPixi.Sketch` instance.
Thanks for pointing out the problem. I'll keep you updated.

**Comment by taranda** [comment]
Awesome.  I just submitted a PR to expo-pixi that partially solves this problem.  However, even with the underlying `<Expo.GLView>` ref exposed, the `Expo.GLView.takeSnapshotAsyc()` method is returning a completely black image on Android.  It works just fine on iOS.

```js
  _save = async () => {
    if (this.props.signature && this.props.signature.uri)
      Expo.FileSystem.deleteAsync(this.props.signature.uri, { idempotent:  true });

    let image = await this.sketch.glView.takeSnapshotAsync();

    this.props.saveSignature({
      uri: image.uri,
      width: image.width,
      height: image.height,
    });
  }
  _onReady = () => { this.setState({ ready: true }); }
  _sketchRef = ref => { this.sketch = ref; }
  _sketchUndo = () => { this.sketch.undo(); }

  render () {
    return (
      <View style={styles.container}>
        <View style={styles.mainSection}>
          <View style={styles.paddingView}/>
          <View style={styles.signatureSection}>
            <ExpoPixi.Sketch
              ref={ this._sketchRef }
              strokeColor={Color.black}
              strokeWidth={5}
              strokeAlpha={1}
              style={styles.sketch}
              onReady={ this._onReady }
            />
          </View>
          <View style={styles.paddingView}/>
          <View style={{ width: '100%', alignItems: 'center' }}>
            <View
              style={{ alignItems: 'center', justifyContent: 'center', flexDirection: 'row', width: '90%' }}
            >
              <Button
                buttonType={ Color.lightGray }
                style={{ flex: 1, width: '100%', marginLeft: 0, borderWidth: 1, borderColor: Color.veryDarkGray }}
                textStyle={{ color: Color.veryDarkGray }}
                onPress={ this._sketchUndo }
              >
                Undo
              </Button>
              <Button
                buttonType={ Color.action }
                style={{ flex: 1, width: '100%', marginRight: 0 }}
                onPress={ this._save }
              >
                Save
              </Button>
            </View>
          </View>
        </View>
      </View>
    );
  }
}
```

I am using the version of `ExpoPixi.Sketch` contained in [PR #15](https://github.com/expo/expo-pixi/pull/15).

I am not sure if I am doing something wrong or if there is a bug in Expo.

**Comment by tsapeta** [comment]
@taranda 
This is definitely a bug, but it seems to be really weird. Generally, snapshots are working on Android but the transparent color is misinterpreted - you can try it out by changing `strokeColor` to something different than black. So far, we don't support transparent snapshots as only JPG format is supported by this new function, but I'm going to add PNG format to our backlog so it should be possible soon.
Until then, you can change `ExpoPixi.Sketch` to draw on opaque background.

So basically these two lines should be changed:
https://github.com/expo/expo-pixi/blob/e95dba3d92bb069c9e31f24501384c239222bd45/lib/components/Sketch.js#L187-L188

to this:
```javascript
backgroundColor: 0xffffff,
transparent: false,
```

Sorry for the inconvenience!

**Comment by taranda** [comment]
That worked. I was half way through converting the Sketch into an SVG by iterating through `Sketch.lines.points`.  This saves me all of that mess.  Thank you.

**Comment by pa-an** [comment]
@tsapeta , I initialized an empty project with this package.json
```{
  "main": "node_modules/expo/AppEntry.js",
  "private": true,
  "dependencies": {
    "expo": "^26.0.0",
    "expo-three": "^2.2.2-alpha.1",
    "react": "16.3.0-alpha.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-26.0.0.tar.gz",
    "three": "^0.91.0"
  }
}
```
Then create a simple App.js with Expo.GLView component, it crashed and out to iphone's home screen. Have I done anything wrong? please help me on this. Thanks

**Comment by tsapeta** [comment]
@pa-an 
I've just tested this example on iPhone 5s (iOS 10.3.1) and Samsung Galaxy S4 (5.0.1) and it works fine on both. Can you tell me more details about devices on which you're testing it?

**Comment by tsapeta** [comment]
@pa-an 
Alright, it seems that iPhone 5 doesn't support OpenGL ES 3.0 which has been introduced in SDK 26, that's why it's crashing. I'm going to fix that in the next release.

Anyway, it should still work on older Androids because they have a fallback to OpenGL ES 2.0. I'll take a deeper look at this. So far I tested it on Nexus 5 with Android 6.0.1 - works fine.

**Comment by pa-an** [comment]
@tsapeta , I got it, thank you very much
Please notice me if you guys have some update.

**Comment by pa-an1** [comment]
@tsapeta I've updated SDK to v27 and thee, expo-three to latest version but iPhone 5 still not working. Android and Iphone6 are ok. I've tested same App.js above.
